server/users.ts

```diff
@@ -587,28 +587,18 @@ export class User extends Chat.MessageContext {
 			Rooms.get(inRoomID)!.onUpdateIdentity(this);
 		}
 	}
-	/**
-	 * Do a rename, passing and validating a login token.
-	 *
-	 * @param name The name you want
-	 * @param token Signed assertion returned from login server
-	 * @param newlyRegistered Make sure this account will identify as registered
-	 * @param connection The connection asking for the rename
-	 */
-	async rename(name: string, token: string, newlyRegistered: boolean, connection: Connection) {
-		let userid = toID(name);
-		if (userid !== this.id) {
-			for (const roomid of this.games) {
-				const room = Rooms.get(roomid);
-				if (!room || !room.game || room.game.ended) {
-					this.games.delete(roomid);
-					console.log(`desynced roomgame ${roomid} renaming ${this.id} -> ${userid}`);
-					continue;
-				}
-				if (room.game.allowRenames || !this.named) continue;
-				this.popup(`You can't change your name right now because you're in ${room.game.title}, which doesn't allow renaming.`);
-				return false;
+	async validateToken(token: string, name: string, userid: ID, connection: Connection) {
+		if (!token && Config.noguestsecurity) {
+			if (Users.isTrusted(userid)) {
+				this.send(`|nametaken|${name}|You need an authentication token to be a trusted user.`);
```

Comment:
```suggestion
				this.send(`|nametaken|${name}|You need an authentication token to log in as a trusted user.`);
```

Reply:
The original error messages reads to me as "you can log in, you just won't be marked as trusted"